src/core/dbus.c

```diff
@@ -139,10 +139,9 @@ static int signal_disconnected(sd_bus_message *message, void *userdata, sd_bus_e
 
         if (bus == m->api_bus)
                 bus_done_api(m);
-        if (bus == m->system_bus)
+        else if (bus == m->system_bus)
```

Comment:
this isn't right. for the system instance the system bus and API bus connection will be the same really, in that case we should call both functions, not just one

Reply:
Indeed. I removed this part.